src/xdp/rx.c

```diff
@@ -1131,6 +1198,62 @@ XdpRxQueueDeregisterNotifications(
     }
 }
 
+NTSTATUS
+XdpRxQueueEnableChecksumOffload(
+    _In_ XDP_RX_QUEUE *RxQueue
+    )
+{
+    NTSTATUS Status;
+
+    TraceEnter(TRACE_CORE, "RxQueue=%p", RxQueue);
+
+    if (RxQueue->FrameExtensionSet == NULL) {
+        RxQueue->IsChecksumOffloadEnabled = FALSE;
+        Status =
+            XdpExtensionSetCreate(
```

Comment:
Same as in the activate function, we should be able to assume the extension sets are not null.

Reply:
So, these changes were added because spinxsk was failing because we were de-referencing a null frame extension set. TLDR; it wasn't enough to just have it be created in `XdpRxQueueCreate`.

Reply:
resolved offline.